### DISCUSSION TITLE
Automatically get title keys from NSP

### DISCUSSION THREAD

**Discussion by MythicalPlayz** [skyline-emu/skyline#121] [Ideas]
I was wondering if it is possible to automatically get the title key from the NSP file using .tik be implemented?

**Comment by timleg002** [comment]
i was working on this, haven't got time to finish it though

**Comment by badge059** [comment]
When are we going to able to see the screen so we can use the emulator? Just wondering:)

**Comment by leopeo-star** [comment]
I was just wondering any idea or guess how long until games can be playable on it if soon enough then what kind if games will we be able to play at first? 
Another question will this be released on playstore/appstore?

**Reply by MythicalPlayz** [reply]
1: If everything goes well then it a matter of months and worst case 1 year.
2: So far not a lot of games boot to menu
3: It will likely be released on Google play once its done.
4: It is Android Exclusive.(why, ask apple)

**Reply by leopeo-star** [reply]
Thank you for replying I can't wait to test it i did forgot to ask one more question will this work on Snapdragon 712 and above or do we require Snapdragon 845 series and above? What are the requirements? And as u said maybe in a matter of month or 2 then will skyline be available for public to use and play as soon as it can emulator games? Or will the beta be only available for few people?

**Reply by timleg002** [reply]
> 1: If everything goes well then it a matter of months and worst case 1 year.
> 2: So far not a lot of games boot to menu
> 3: It will likely be released on Google play once its done.
> 4: It is Android Exclusive.(why, ask apple)

AFAIK, though, correct me if I'm wrong, it's possible to port this to iOS (and probably Macs with ARM M1 chips) but it would require a lot of work, mainly the graphics that would be needed to be ported  to Apple's proprietary graphic backend, Metal (There's a tool that could convert would-be Skyline's graphic backend written in Vulkan to Metal) and some of iOS' quirks and other stuff. 
Not mentioning the general "pain" of installation on Apple devices, that would require jailbreaking, and fetching the app each 7 (or 14?) days, because the certificates expiring.
I may have missed a lot of stuff here, so there's that ¯\\_(ツ)_/¯

**Reply by MythicalPlayz** [reply]
Lets answer both questions.
Any chip would run Skyline tho We say snapdragon 845 for minimum good performance 
____________
The main reason is that it won't come to apple is that
Vulkan isn't really supported by Apple.
Even if it is you would still need a jailbreaked Iphone as apple can't let you access storage.
(unlike android)
While theoricaly it is possible that the emulator could have an IOS port as it is made in C++ while ,Koltin is just for app design.
It will not be worth it.

I am not mark so this could be wrong.

**Reply by timleg002** [reply]
Yeah, uh, that's kinda what I said. The code could be ported with a good chance, even if it was written in Kotlin (due to Kotlin's ability to compile to iOS native binaries, if not too reliant on Java/JVM intrinsics.)

**Comment by badge059** [comment]
Thank you very much for letting me know have a great day:)

**Comment by htet12345** [comment]
hi

**Comment by PixelyIon** [comment]
This was resolved in #138, locking further discussion.
